### PR TITLE
Fav list

### PR DESCRIPTION
**Pull request by SteveVaknin**
image

Added `mobx` store. 

For now the favorites are the first 2 jobs from the db 😅

### CODE REVIEW

File: src/components/Application-Header/favorite-list/FavoriteList.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                    vertical: 'bottom',
+                    horizontal: 'center',
+                  }}
+                  transformOrigin={{
+                    vertical: 'top',
+                    horizontal: 'center',
+                  }}
+                onClose={handleClose}
+            >
+                {jobsStore.advertisements.slice(0,2).map((ad) => (
```
Comment (JetoPistola) [inline src/components/Application-Header/favorite-list/FavoriteList.tsx:52]: `2` is a magic number ^^

File: src/components/Application-Body/JobsAppBody.tsx [hunk 2]
```diff
@@ -31,40 +32,43 @@ const styles = makeStyles({
...
-            fetchAllAdsAfterPost={fetchAllAds} />
-        <JobsList ads={getFilteredAds()} isFiltered={searchValue !== '' && ads.length !== 0}/>
-    </div>
-  );
+    return (
+        <div className={classes.appBodyContent}>
+            <Header 
+                searchValue={searchValue} 
+                onSearchValueChange={onSearchValueChange}
+                fetchAllAdsAfterPost={() => {jobsStore.loadAdvertisements()}} />
```
Comment (JetoPistola) [inline src/components/Application-Body/JobsAppBody.tsx:68]: Refererence to method doesn't work?

fetchAllAdsAfterPost={jobsStore.loadAdvertisements}

File: src/components/Application-Body/JobsAppBody.tsx [hunk 3]
```diff
@@ -31,40 +32,43 @@ const styles = makeStyles({
...
+    useEffect(() => {
+        jobsStore.loadAdvertisements();
+    }, []);
 
     const onSearchValueChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
         setSearchValue(event.target.value);
     }
 
     const getFilteredAds = (): Advertisement[] => {
+        const ads = jobsStore.advertisements;
```
Comment (stavbi1) [inline src/components/Application-Body/JobsAppBody.tsx:54]: `: Advertisment[]`

File: src/components/Application-Body/JobsAppBody.tsx [hunk 4]
```diff
@@ -31,40 +32,43 @@ const styles = makeStyles({
...
+interface JobsAppBodyOwnProps {
+    jobsStore?: JobsStore;
+}
 
-  const fetchAllAds = (): void => {
-    getAllAds().then(data => setAds(data));
-  }
+const JobsAppBody: React.FC<JobsAppBodyOwnProps> = (props): JSX.Element => {
+    const classes = styles({});
+    const jobsStore = props.jobsStore!;
```
Comment (stavbi1) [inline src/components/Application-Body/JobsAppBody.tsx:41]: The store dosen't have a type?

File: src/store/JobsStore.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    get isLoading () {
+        return this.isLoadingAds.get();
+    }
+
+    loadAdvertisements = () => {
+        // Fetch data from server
+        this.isLoadingAds.set(true);
+        getAllAds().then(data => {
```
Comment (stavbi1) [inline src/store/JobsStore.ts:20]: async/await

File: src/components/Application-Header/favorite-list/FavoriteList.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+interface FavoriteListProps extends WithStyles<typeof styles> {
+    jobsStore?: JobsStore;
+}
+
+const FavoriteList: React.FC<FavoriteListProps> = (props): JSX.Element => {
+    const { classes } = props;
+    const jobsStore = props.jobsStore!;
+
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (stavbi1) [inline src/components/Application-Header/favorite-list/FavoriteList.tsx:24]: maybe change to a more meaningful name

File: src/components/Application-Header/favorite-list/FavoriteList.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+interface FavoriteListProps extends WithStyles<typeof styles> {
+    jobsStore?: JobsStore;
+}
+
+const FavoriteList: React.FC<FavoriteListProps> = (props): JSX.Element => {
+    const { classes } = props;
+    const jobsStore = props.jobsStore!;
+
+    const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```
Comment (stavbi1) [inline src/components/Application-Header/favorite-list/FavoriteList.tsx:23]: I see that your'e only using the anchor to determine whether to display the popover or not, IMO, it's a bit vague and boolean named `isPopoverOpen` would be great.
But it's up to you.

File: src/components/Application-Header/favorite-list/FavoriteList.tsx (hunk 1 above)
Comment (SteveVaknin) [reply src/components/Application-Header/favorite-list/FavoriteList.tsx:52]: this is just for dev...
We will need to fetch the current user's favs from the server

File: src/components/Application-Header/favorite-list/FavoriteList.tsx (hunk 7 above)
Comment (SteveVaknin) [reply src/components/Application-Header/favorite-list/FavoriteList.tsx:23]: it's not only for open/close flag, it holds the location in which the popover will open (based on the mouse click position I think).

**Pull request merged**